python/dolfinx/fem/assemble.py

```diff
@@ -133,11 +133,19 @@ def assemble_scalar(M: FormMetaClass, constants=None, coeffs=None):
 # -- Vector assembly ---------------------------------------------------------
 
 @functools.singledispatch
-def assemble_vector(L: FormMetaClass, constants=None, coeffs=None) -> la.VectorMetaClass:
+def assemble_vector(L: typing.Any, constants=None, coeffs=None):
```

Comment:
Is `Any` really what we want?

Reply:
Probably not. It could be a union of all allowed types.